packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -761,75 +666,53 @@ function getLeafNodeFromJSObject(tree: Jsonable): Jsonable {
 	}
 }
 
-function readCursorTree(
-	forest: IForestSubscription,
-	numberOfNodes: number,
-	shape: TreeShape,
-	dataType: string,
-) {
+function readCursorTree(forest: IForestSubscription, numberOfNodes: number, shape: TreeShape) {
 	const readCursor = forest.allocateCursor();
 	moveToDetachedField(forest, readCursor);
 	assert(readCursor.firstNode());
 	let nodesRead = 0;
-	let currentTotal = 0 as number | string | boolean | JsonableTree;
+	let currentTotal = 0 as number;
 	switch (shape) {
 		case TreeShape.Deep:
 			for (let i = 0; i < numberOfNodes; i++) {
-				assert(readCursor.firstField());
-				assert(readCursor.firstNode());
-				currentTotal = applyOperationDuringRead(currentTotal, readCursor.value, dataType);
+				readCursor.enterField(localFieldKey);
+				readCursor.enterNode(0);
+				currentTotal = applyOperationDuringRead(currentTotal, readCursor.value);
 			}
 			break;
 		case TreeShape.Wide:
-			assert(readCursor.firstField());
+			readCursor.enterField(localFieldKey);
 			for (let inNode = readCursor.firstNode(); inNode; inNode = readCursor.nextNode()) {
 				nodesRead += 1;
-				currentTotal = applyOperationDuringRead(currentTotal, readCursor.value, dataType);
+				currentTotal = applyOperationDuringRead(currentTotal, readCursor.value);
 			}
 			assert(nodesRead === numberOfNodes);
 			break;
 		default:
 			throw new Error("unreachable case");
 	}
 	readCursor.free();
+	return currentTotal;
 }
 
-function applyOperationDuringRead(
-	current: number | string | boolean | JsonableTree,
-	value: Value,
-	dataType: string,
-) {
+function applyOperationDuringRead(current: number, value: Value) {
 	assert(value !== undefined);
```

Comment:
This assert just makes this function not work when there is no value. I think that makes this function less useful.
If its always used when there is a value, and these tests only include number values, its prabably better to jusr remove this function and replaces its use with `sum += value as number;`.

If there really are other cases when this is called, its probably better to remove this assert (that way it actually works for everything its signature says you can pass in)